qt/scientific_interfaces/Inelastic/Processor/ElwinPresenter.h

```diff
@@ -68,35 +71,47 @@ class MANTIDQT_INELASTIC_DLL ElwinPresenter : public DataProcessor, public IElwi
   void handleRowModeChanged() override;
   void updateAvailableSpectra() override;
 
-  void setInputWorkspace(MatrixWorkspace_sptr inputWorkspace);
+  void setInputWorkspace(const MatrixWorkspace_sptr &inputWorkspace);
   virtual void setSelectedSpectrum(int spectrum);
   int getSelectedSpectrum() const;
-  MatrixWorkspace_sptr getInputWorkspace() const;
+  MatrixWorkspace_sptr getInputWorkspace() const override;
   MatrixWorkspace_sptr getPreviewPlotWorkspace();
   void setPreviewPlotWorkspace(const MatrixWorkspace_sptr &previewPlotWorkspace);
 
+  void connectObservers() const;
+  void disconnectObservers() const;
+  void removeWorkspace(std::string const &workspaceName) const override;
+
 protected:
   void runComplete(bool error) override;
-  void newInputDataFromDialog();
   virtual void addDataToModel(MantidWidgets::IAddWorkspaceDialog const *dialog);
 
 private:
+  void updateInterface();
   void updateTableFromModel();
   void updateIntegrationRange();
 
   std::vector<std::string> getOutputWorkspaceNames();
   std::string getOutputBasename();
   bool checkForELTWorkspace();
-  void newPreviewFileSelected(const std::string &workspaceName, const std::string &filename);
-  void newPreviewWorkspaceSelected(const std::string &workspaceName);
-  WorkspaceID findWorkspaceID();
+  void newPreviewWorkspaceSelected(int index);
+  std::optional<WorkspaceID> findWorkspaceID(std::string const &name) const;
+
+  void handleRemoveEvent(Mantid::API::WorkspacePreDeleteNotification_ptr pNf);
+  void handleClearEvent(Mantid::API::ClearADSNotification_ptr pNf);
+  void handleRenameEvent(Mantid::API::WorkspaceRenameNotification_ptr pNf);
 
   IElwinView *m_view;
   std::unique_ptr<IElwinModel> m_model;
   std::unique_ptr<IDataModel> m_dataModel;
   int m_selectedSpectrum;
   std::weak_ptr<MatrixWorkspace> m_previewPlotWorkspace;
   MatrixWorkspace_sptr m_inputWorkspace;
+
+  /// Poco Observers for ADS Notifications
+  Poco::NObserver<ElwinPresenter, Mantid::API::WorkspacePreDeleteNotification> m_remObserver;
+  Poco::NObserver<ElwinPresenter, Mantid::API::ClearADSNotification> m_clearObserver;
+  Poco::NObserver<ElwinPresenter, Mantid::API::WorkspaceRenameNotification> m_renameObserver;
```

Comment:
Would it be possible to get the Presenter to inherit the `AnalysisDataServiceObserver` class and implement the relevant functions so we can avoid using `Poco` in these files? https://github.com/mantidproject/mantid/blob/50916dd41b7bfd467d17aa50d8546998aec951d0/Framework/API/inc/MantidAPI/AnalysisDataServiceObserver.h#L38

Reply:
I was not aware of this class. Thanks for the tip.